core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -595,14 +595,24 @@ thread_manager_start_routine(void *arg)
 
     /* Routine exit */
 
-    /* Detach the native thread here to ensure the resources are freed */
-    wasm_cluster_detach_thread(exec_env);
 #if WASM_ENABLE_DEBUG_INTERP != 0
     wasm_cluster_thread_exited(exec_env);
 #endif
 
+    os_mutex_lock(&cluster_list_lock);
```

Comment:
Is this lock required by `wasm_cluster_del_exec_env`?
Had better update the comment for `wasm_cluster_del_exec_env`:

``` C
/* The caller should lock cluster->lock and 
    cluster_list_lock for thread safety */
```

Reply:
No, this lock here is to lock
```C
if (exec_env->wait_count == 0 && !exec_env->thread_is_detached) {
    ...
}
```

wasm_cluster_del_exec_env only needs to lock cluster->lock